sql/core/src/main/scala/org/apache/spark/sql/execution/PartitionedFileUtil.scala

```diff
@@ -69,8 +69,10 @@ object PartitionedFileUtil {
         b.getHosts -> (b.getOffset + b.getLength - offset).min(length)
 
       // The fragment ends at a position within this block
-      case b if offset <= b.getOffset && offset + length < b.getLength =>
-        b.getHosts -> (offset + length - b.getOffset).min(length)
+      case b if offset <= b.getOffset &&
```

Comment:
I think this might properly be `<` instead, but, I also think it's not needed. The `offset >= b.getOffset` case is already handled above if the fragment starts within the block. And if it starts after the block, then it won't end in the block, and that is checked below. So this should already only handle cases where it starts before the block.

Reply:
Ok，fixed it.

Reply:
I'm saying this clause is not necessary at all.